### PR TITLE
Add command-line and settings elements to XML

### PR DESCRIPTION
**Pull request by CharliePoole**
This adds both a command-line element and a settings element to the XML. The command-line element shows what was entered and the settings element shows the settings pulled from that command line.

However, at this point, the engine defaults have not yet been added to the settings as discussed in issue #713. Only those settings specified on the command-line are included.

### CODE REVIEW

**Comment by rprouse** [comment]
Looks good to me so far.

**Comment by CharliePoole** [comment]
Latest commit adds an EffectiveSettings property. It uses the same settings as the package and then adds a few at their default value. While this could be extended to cover all defaults, it has one fundamental flaw and one lesser flaw.

The serious flaw is this: NumberOfTestWorkers is actually a value that is known to the nunit framework 3.0. It is neither known to or used by the engine. The console runner understands this setting and can set it and pass it to the engine but the engine does not know anything about it. Except now it does: it knows the name and what it's used for as well as the algorithm for calculating the default. Seems a bit much, especially since _it only has meaning for the NUnit 3.0 framework_. Any other framework might not support parallel threads (NUnit v2 for example) or might use a different default.

The default for process model illustrates a lesser problem: The engine now has duplicate code for figuring out the default. This can probably be fixed by refactoring but it may involve a bit of work if we try to cover all default values.

Advice is needed here. How should we deal with the default values for framework settings?
The problem is that the logic that sets the default values for reporting is entirely separate from the logic of the engine itself. If we changed the reported default for NumberOfTestWorkers here, the engine would still use it's own default value. This is not a good design.

I see two ways to go here:

1. Put code into the engine that explicitly replaces defaults in the package before executing it. That's not how we now do it, because we wanted to retain the knowledge that we are using a default for reporting purposes. However, we get this in the XML since the command-line is now included.

2. Create some internal class that returns default values for any package setting and have the engine use it both for reporting and for execution purposes.

**Comment by CharliePoole** [comment]
Some extra text got left in the last comment. I am editing it.

**Comment by CharliePoole** [comment]
Latest merge is mostly comments explaining what all the settings are.

@rprouse Since you are pretty familiar now with the engine, it would help if you could review those changes for accuracy. Feel free to update any typos online.

This commit is in preparation for having the engine add more default settings directly to the test package.

**Comment by CharliePoole** [comment]
Now ready to merge once it passes.

I added all settings, including those set by default, to the XML and then used the XML to produce the same output on the console. Console runner now no longer displays options entered before execution, since any such options will be reflected in the settings report.

I also fixed issue #847 by adding new --process:InProcess and --inprocess options.

**Comment by CharliePoole** [comment]
The last commit for #713 surprised me by affecting NUnitLite. I'm going to back it out and do that as a separate PR.

**Comment by CharliePoole** [comment]
Did a forced update to remove the commit referenced in the last two comments. I thought #847 was a quick fix that could easily be included with this one, but it turns out not to be so.

Can be merged once the CI passes.

File: src/NUnitEngine/nunit.engine/Runners/MasterTestRunner.cs [hunk 1]
```diff
@@ -38,6 +39,45 @@ public class MasterTestRunner : AbstractTestRunner, ITestRunner
...
+                {
+                    _effectiveSettings = new Dictionary<string, object>();
+                    foreach (string key in TestPackage.Settings.Keys)
+                        _effectiveSettings.Add(key, TestPackage.Settings[key]);
+
+                    bool processModelSpecified = _effectiveSettings.ContainsKey(PackageSettings.ProcessModel);
+                    string processModel = processModelSpecified
+                        ? (string)_effectiveSettings[PackageSettings.ProcessModel]
+                        : TestPackage.SubPackages.Count > 1
+                            ? "Multiple"
```
Comment (rprouse) [inline src/NUnitEngine/nunit.engine/Runners/MasterTestRunner.cs]: I won't ask for a change here, but I would like to suggest a change to our coding standard to disallow nested ternary conditional operators. This one isn't too bad, but we have a few in the code that are hard to grok.

Comment (CharliePoole) [inline src/NUnitEngine/nunit.engine/Runners/MasterTestRunner.cs]: I think the standard should probably say "don't use it unless it makes the code easier to read."

In this case, it did - you should see how involved it got before I did this!

**Pull request merged**